src/main/java/org/jboss/tm/usertx/TransactionListener.java

```diff
@@ -0,0 +1,16 @@
+package org.jboss.tm.usertx;
+
+import javax.transaction.Transaction;
+
+/**
+ * A listener callback for transaction related events
+ */
+public interface TransactionListener {
+    /**
+     * The callback for notifying registered listeners of a transaction related event
+     *
+     * @param transaction the transaction that the caller is registering interest in
+     * @param transactionEvent indication of what kind of change has occurred
+     */
+    void stateChange(Transaction transaction, TransactionEvent transactionEvent);
```

Comment:
OK, I can see you could argue we need the transaction.

In terms of the enum issue:
public class TransactionEvent {

    private Transaction transaction;

   private TransactionState transactionState;

    public getTransaction() {
        return transaction;
    }

    public TransactionState getTransactionState() {
         return transactionState;
    }

    public enum TransactionState {
    /**
     * a transaction is associated with the current thread
     */
    ASSOCIATED,

    /**
     * a transaction is just about to be disassociated from the current thread. This
     * can happen because the thread is about to commit, roll back or suspend the
     * transaction that is currently associated with the thread.
     */
    DISASSOCIATING
    }
}